src/track.c

```diff
@@ -203,23 +221,47 @@ void tracking_channel_update(u8 channel)
   {
     case TRACKING_RUNNING:
     {
-      chan->update_count++;
       chan->sample_count += chan->corr_sample_count;
+      chan->code_phase_early = (u64)chan->code_phase_early +
+                               (u64)chan->corr_sample_count
+                                 * chan->code_phase_rate_fp_prev;
+      chan->carrier_phase += (s64)chan->carrier_freq_fp_prev
+                               * chan->corr_sample_count;
+      /* TODO: Fix this in the FPGA - first integration is one sample short. */
+      if (chan->update_count == 0)
+        chan->carrier_phase -= chan->carrier_freq_fp_prev;
+      chan->code_phase_rate_fp_prev = chan->code_phase_rate_fp;
+      chan->carrier_freq_fp_prev = chan->carrier_freq_fp;
 
       /* TODO: check TOW_ms = 0 case is correct, 0 is a valid TOW. */
       if (chan->TOW_ms > 0) {
         /* Have a valid time of week. */
-        chan->TOW_ms++;
-        if (chan->TOW_ms == 7*24*60*60*1000)
-          chan->TOW_ms = 0;
+        chan->TOW_ms += chan->short_cycle ? 1 : (chan->int_ms-1);
+        chan->TOW_ms %= 7*24*60*60*1000;
+      }
 
-        /* Turn off FLL aiding. For now we do this here because having a valid
-         * TOW is a very good indication that the tracking loops have locked. */
-        chan->tl_state.carr_filt.aiding_igain = 0;
```

Comment:
No.  When we switch to 20ms integration the loop filter is reinitialised with an FLL gain of 0.  The PLL filter coefficients also need to be recalculated, to aided_tl_init() is called with the new parameters.